data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraClientTest.java

```diff
@@ -0,0 +1,120 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.jira.JiraClient;
+import org.opensearch.dataprepper.plugins.source.jira.JiraIterator;
+import org.opensearch.dataprepper.plugins.source.jira.JiraService;
+import org.opensearch.dataprepper.plugins.source.jira.JiraSourceConfig;
+import org.opensearch.dataprepper.plugins.source.jira.rest.auth.JiraAuthConfig;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.CrawlerSourceConfig;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.PluginExecutorServiceProvider;
+import org.opensearch.dataprepper.plugins.source.source_crawler.coordination.state.SaasWorkerProgressState;
+import org.springframework.web.client.RestTemplate;
+
+import java.lang.reflect.Field;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class JiraClientTest {
+    @Mock
+    private JiraAuthConfig authConfig;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private Buffer<Record<Event>> buffer;
+
+    @Mock
+    private SaasWorkerProgressState saasWorkerProgressState;
+
+    @Mock
+    private CrawlerSourceConfig crawlerSourceConfig;
+
+    @Mock
+    private JiraSourceConfig jiraSourceConfig;
+
+    @Mock
+    private JiraService jiraService;
+
+    @Mock
+    private JiraIterator jiraIterator;
+
+    private PluginExecutorServiceProvider executorServiceProvider = new PluginExecutorServiceProvider();
+    private JiraClient jiraClient;
+
+    @BeforeEach
+    void setUp() throws JsonProcessingException {
+        jiraClient = new JiraClient(jiraService, jiraIterator, executorServiceProvider, jiraSourceConfig);
+    }
+
+    @Test
+    void testListItems() {
+        assertNotNull(jiraClient.listItems());
+    }
+
+    @Test
+    void testSetLastPollTime() throws NoSuchFieldException, IllegalAccessException {
+        jiraClient.setLastPollTime(Instant.ofEpochSecond(1234L));
+        Field pollTime = jiraClient.getClass().getDeclaredField("lastPollTime");
+        pollTime.setAccessible(true);
+        Object oldPollTime = pollTime.get(jiraClient);
+        jiraClient.setLastPollTime(Instant.ofEpochSecond(5678L));
```

Comment:
Let's add the setter method or relax the access constraints in the class itself instead of reflection

Reply:
call setter